src/envoy/utils/mixer_control.h

```diff
@@ -42,5 +48,40 @@ Grpc::AsyncClientFactoryPtr GrpcClientFactoryForCluster(
     const std::string &cluster_name, Upstream::ClusterManager &cm,
     Stats::Scope &scope);
 
+std::unique_ptr<const LocalAttributes> CreateLocalAttributes(
```

Comment:
Do you expect these functions to be used by multiple places? if no, just one place, please just move code to that place.